### DISCUSSION TITLE
Is it possible to support custom S3 host to support self-built S3 service?

### DISCUSSION THREAD

**Discussion by KeyAI** [Xheldon/Notion-Flow-Prod#2] [Ideas]
support self-built S3 service

**Comment by KeyAI** [comment]
@Xheldon

**Reply by Xheldon** [reply]
@KeyAI  The main focus of this extension is currently on AI and plugin directions, of course, I'd be happy to add a feature on the fly—if it's easy to add.
Regarding your self-hosted S3 service, do you have any documentation I can refer to, such as how to authorize access to your own S3 service, etc.?

**Comment by KeyAI** [comment]
In my case, I built my own OSS using [Cloudflare R2](https://www.cloudflare.com/developer-platform/r2/),  its API is compatible with S3

**Reply by Xheldon** [reply]
Sure, I will look into the documentation. If it's feasible, I will add that feature within the next two versions. Thanks for the suggestion!

**Reply by Xheldon** [reply]
@KeyAI  If my research is correct, [this document](https://developers.cloudflare.com/r2/examples/aws/aws-sdk-js-v3/) may be able to resolve your issue. Your self-hosted OSS needs to provide an additional endpoint beyond the four values provided by AWS S3. If that's not a problem, implementing this function seems quite straightforward, and I will complete it as soon as possible.

However, there is an issue - I haven't used the R2 service, so when I build a development version, would you be willing to help me test its usability? Thank you!

**Comment by Xheldon** [comment]
Done.

Notion Flow `0.4.1` has been submitted for review.
